test/benchmarks.cc

```diff
@@ -0,0 +1,44 @@
+#include <catch2/catch.hpp>
+
+#include <intrusive_graph.h>
+
+using namespace celerity::detail;
+
+struct bench_graph_node : intrusive_graph_node<bench_graph_node> {};
+
+TEST_CASE("benchmark intrusive graph dependency handling", "[benchmark]") {
+	// this is deliberately chosen to be relatively small, since that's what we expect to encounter in practice
+	int N = GENERATE(2, 7, 13);
+
+	BENCHMARK("creating and adding dependencies") {
+		bench_graph_node n0;
+		bench_graph_node nodes[N];
```

Comment:
As discussed with @PeterTh, this only works because we have GNU extensions and should probably be a `std::vector`.